File: src/ServerPulse.Frontend/src/app/modules/shared/domain/dtos/statistics/serverStatisticsResponse.ts
import { TimeSpan } from "../../..";

export interface ServerStatisticsResponse {
    isAlive: boolean;
    dataExists: boolean;
    serverLastStartDateTimeUTC: Date | null;
    serverUptime: TimeSpan | null;
    lastServerUptime: TimeSpan | null;
    lastPulseDateTimeUTC: Date | null;
    collectedDateUTC: Date,
    isInitial: boolean;
}

export function convertToServerStatisticsResponse(data: any): ServerStatisticsResponse {
    return {
        isAlive: data?.IsAlive,
        dataExists: data?.DataExists,
        serverLastStartDateTimeUTC: data.ServerLastStartDateTimeUTC ? new Date(data.ServerLastStartDateTimeUTC) : null,
        serverUptime: data.ServerUptime ? TimeSpan.fromString(data.ServerUptime) : null,
        lastServerUptime: data.LastServerUptime ? TimeSpan.fromString(data.LastServerUptime) : null,
        lastPulseDateTimeUTC: data.LastPulseDateTimeUTC ? new Date(data.LastPulseDateTimeUTC) : null,
        collectedDateUTC: data?.CollectedDateUTC,
        isInitial: data?.IsInitial
    };
}

File: src/ServerPulse.Backend/AnalyzerApi/Domain/Dtos/Responses/ServerStatisticsResponse.cs
﻿namespace AnalyzerApi.Domain.Dtos.Responses
{
    public class ServerStatisticsResponse
    {
        public bool IsAlive { get; set; }
        public bool DataExists { get; set; }
        public DateTime? ServerLastStartDateTimeUTC { get; set; }
        public TimeSpan? ServerUptime { get; set; }
        public TimeSpan? LastServerUptime { get; set; }
        public DateTime? LastPulseDateTimeUTC { get; set; }
        public DateTime CollectedDateUTC { get; set; }
        public bool IsInitial { get; set; }
    }
}